### PR TITLE
Feat/trainer

### PR DESCRIPTION
**Pull request by deep-diver**
This is a draft PR, and there is an issue that I am working on:
- in the line 82(`upsampled_logits = tf.image.resize(...`), I get an error `AttributeError: Tensor object has no attribute 'size'.`

Important Note:
- I am experimenting TFX pipeline locally before Vertex AI, so small batch of data is included under `/data` directory.

To reproduce:
- Checkout `feat/trainer` branch
- Install required dependencies with `training_pipeline/requirements.txt`
- Run the following three commands in order
  - `tfx pipeline create --pipeline-path=local_runner.py --engine=local`
  - `tfx pipeline compile --pipeline-path=local_runner.py --engine=local`
  - `tfx run create --pipeline-name=segformer-training-pipeline --engine=local`

if you think I should write a notebook rather than scripts, please let me know.

### CODE REVIEW

File: training_pipeline/models/model.py [hunk 1]
```diff
@@ -1,266 +1,229 @@
...
-            tf.TensorSpec(
-                shape=[None, 224, 224, 3],
-                dtype=tf.float32,
-                name=_transformed_name(_IMAGE_KEY),
-            )
+        upsampled_logits = tf.image.resize(
+            logits,
+            images.size[
+                ::-1
+            ],  # We reverse the shape of `image` because `image.size` returns width and height.
```
Comment (sayakpaul) [inline training_pipeline/models/model.py:79]: TensorFlow tensors don't have any `size` attribute. Use `tf.shape()` or `.shape`.

**Review by sayakpaul — CHANGES_REQUESTED** [review]

**Comment by deep-diver** [comment]
`preprocess` part will be updated when the adjusted TFRecords for UNET is ready.

**Comment by sayakpaul** [comment]
@deep-diver is this PR ready to review?

**Comment by deep-diver** [comment]
@sayakpaul 

will let you know after reflecting the changes in TFRecord

**Comment by deep-diver** [comment]
@sayakpaul 

ready for review

**Comment by sayakpaul** [comment]
Thanks. 

Will review tomorrow.

**Pull request closed without merging**